gradle-android-command-plugin/src/main/groovy/com/novoda/gradle/command/Monkey.groovy

```diff
@@ -6,30 +6,49 @@ class Monkey extends AdbTask {
 
     def events
     def seed
+    def categories
 
     protected handleCommandOutput(def text) {
         super.handleCommandOutput(text)
-        if (text.contains("Monkey aborted"))
+        if (text.contains("Monkey aborted")) {
             throw new GroovyRuntimeException('Monkey run failed')
+        }
     }
 
     @TaskAction
     void exec() {
-        def arguments = ['shell', 'monkey', '-p', packageName, '-v', getEvents()]
-        if (getSeed())
+        def arguments = ['shell', 'monkey']
+        arguments += ['-p', packageName]
+        arguments += getFormattedCategories()
+        arguments += ['-v', getEvents()]
+        if (getSeed()) {
             arguments += ['-s', getSeed()]
+        }
         assertDeviceAndRunCommand(arguments)
     }
 
+    def getFormattedCategories() {
+        getCategories().collect { "-c " + it }
```

Comment:
Yes, just iterate over the categories, but return and array of distinct (whitespace-free) parameters instead of joining them here. 
Looking at this code again, I see that you treat `categories` as a list usually, but give it the default of an empty string. Setting the default to empty list (`[]`) seems more natural and might avoid setting `-c` at all when no categories are specified.